BD-4776: Update example for Context step (#9686)
braze-inc/braze-docs: _docs/_user_guide/engagement_tools/canvas/canvas_components/context.md

## Using context variables {#using-context-variables}

> For example, let's say you want to notify passengers about their VIP lounge access before their upcoming flight. This message should only be sent to passengers who purchased a first-class ticket. Whereas membership status is something you'd track on the user profile, a first-class ticket purchase is more temporary. A context variable is a flexible way to track this information.

Author: ```suggestion
For example, let's say you want to notify passengers about their VIP lounge access before their upcoming flight. This message should only be sent to passengers who purchased a first-class ticket. A context variable is a flexible way to track this information.
```

> Users will enter the Canvas when they purchase a plane ticket. To determine lounge access eligibility, we'll create a context variable called `lounge_access_granted` in a Context step, then reference that context variable in subsequent steps of the user journey.

Reviewer: Thanks I think it's clearer with the changes. One nit is that `lounge_access_granted` sounds like a boolean but the examples uses it as the booking class and does string comparisons against it in the liquid.

Author: Thanks! I can make some edits to the example to use `lounge_access` instead if that helps deviate away from the boolean scenario 👍

Reviewer: Maybe a better example is to make the context variable be dependent on the custom_attribute.Purchased_Flight, so:

lounge_access_granted: Boolean
{% if custom_attribute.Purchased_flight %}
true
elseif
false
endif

(I'm not a liquid expert so will probably need some changing)

That way the context variable is doing a meaningful manipulation rather than passing the custom attribute directly to the message step. Then in the message step you can use

```
if context.lounge_access_granted
  Enjoy exclusive VIP access
  ...
```

Author: Just pushed the updates for a more meaningful example!

![Context variable set up to track if a passenger qualifies for VIP lounge access.][5]{: style="max-width:90%"}

> In this Context step, we'll use {% raw %}`{{custom_attribute.${Purchased Flight}}}`{% endraw %} to determine if the type of flight they've purchased is first class.

Reviewer 2: ```suggestion
In this Context step, we'll use {% raw %}`{{custom_attribute.${purchased_flight}}}`{% endraw %} to determine if the type of flight they've purchased is first class.
```
I think!

> In this Context step, we'll use the Liquid value {% raw %}`{{custom_attribute.${Purchased Flight}}}`{% endraw %} to determine if the type of flight they've purchased is first class.

Author: ```suggestion
In this Context step, we'll use binary logic to set {% raw %}`{{custom_attribute.${Purchased Flight}}}`{% endraw %} to `true` to determine if the type of flight they've purchased is first class.
```

> In this Context step, we'll use binary logic to set {% raw %}`{{custom_attribute.${Purchased Flight}}}`{% endraw %} to `true` to determine if the type of flight they've purchased is first class.

Author: ```suggestion
In this Context step, we'll use {% raw %}`{{custom_attribute.${Purchased Flight}}}`{% endraw %} to determine if the type of flight they've purchased is first class.
```

> In this Context step, we'll use the Liquid value {% raw %}`{{custom_attribute.${Purchased Flight}}}`{% endraw %} to determine what type of flight they've purchased (first class, business, or economy).

Reviewer: > we'll use the Liquid value {% raw %}`{{custom_attribute.${Purchased Flight}}}`{% endraw %}

This seemingly conflicts with line 97 which says that the purchase flight class isn't on the user profile but the liquid is accessing a custom attribute.

> Whereas membership status is something you'd track on the user profile, a first-class ticket purchase is more temporary. A context variable is a flexible way to track this information.

Author: Removed the sentence for clarity

- The user is a Rewards Club member (checked against their user profile attribute)
- The user has purchased a first-class ticket (determined from the Canvas entry event data)

> Next, we'll create a short delay to target users where {% raw %}`{{context.${lounge_access_granted}}}`{% endraw %} is `true`. Then, we'll set up a Message step to create a push notification that includes personalized lounge information for these users. 

Reviewer: I think this still needs some updating. 

The docs page still has a graphic of the audience step  https://www.braze.com/docs/user_guide/engagement_tools/canvas/canvas_components/context#using-context-variables 
![image](https://github.com/user-attachments/assets/d23ce8f9-3734-4674-8ed4-355a29551edd)

>Next, we'll create a short delay to target users where {% raw %}`{{context.${lounge_access_granted}}}`{% endraw %} is `true`.
This comparison of a variable to a boolean is still technically a decision/audience split step. One suggestion I have is to update this to a 2-step example sequence instead, where there can be 2 images per step
1. Using a separate delay step with personalization 
- short delay Is this a personalized delay step? https://www.braze.com/docs/user_guide/engagement_tools/canvas/canvas_components/delay_step#personalized-delays
- message step that uses a context variable with liquid.
